lib/daru/io/exporters/csv.rb

```diff
@@ -37,30 +37,45 @@ class CSV < Base
         #   #  1   2   4
         #
         #   Daru::IO::Exporters::CSV.new(df, "filename.csv", convert_comma: true).call
-        def initialize(dataframe, path, converters: :numeric, headers: nil,
-          convert_comma: nil, **options)
+        def initialize(dataframe, path, converters: :numeric, compression: false,
+          headers: nil, convert_comma: nil, **options)
           require 'csv'
 
           super(dataframe)
           @path          = path
           @headers       = headers
+          @compression   = compression
           @convert_comma = convert_comma
           @options       = options.merge converters: converters
         end
 
         def call
-          writer = ::CSV.open(@path, 'w', @options)
-          writer << @dataframe.vectors.to_a unless @headers == false
+          contents = process_dataframe
 
-          @dataframe.each_row do |row|
-            writer << if @convert_comma
-                        row.map { |v| v.to_s.tr('.', ',') }
-                      else
-                        row.to_a
-                      end
+          if @compression == :gzip
+            ::Zlib::GzipWriter.open(@path) do |gz|
+              contents.each { |content| gz.write("#{content.join(',')}\n") }
+              gz.close
+            end
+          else
+            csv = ::CSV.open(@path, 'w', @options)
+            contents.each { |content| csv << content }
+            csv.close
+          end
+        end
+
+        private
+
+        def process_dataframe
+          order = [@dataframe.vectors.to_a] unless @headers == false
+          data  = @dataframe.map_rows do |row|
+            next row.to_a unless @convert_comma
+            row.map { |v| v.to_s.tr('.', ',') }
```

Comment:
@athityakumar doesn't the Ruby standard library have a CSV writer that you can use out of the box? I'm sure they must have handled these problems already.